src/snowflake/snowpark/dataframe.py

```diff
@@ -5287,7 +5306,10 @@ def random_split(
             if seed:
                 ast.seed.value = seed
             if statement_params:
-                ast.statement_params = statement_params
+                for k, v in statement_params.items():
```

Comment:
Reuse ast_utils.build_expr_from_dict_str_str
```suggestion
                build_expr_from_dict_str_str(ast.statement_params, statement_params)
```